openwisp_firmware_upgrader/base/models.py

```diff
@@ -250,23 +250,11 @@ def clean(self):
             raise ValidationError({'type': 'Could not find boards for this type'})
 
     def delete(self, *args, **kwargs):
+        """
+        Delete only deletes the database record
+        The actual file deletion is handled by the signal handlers
+        """
         super().delete(*args, **kwargs)
-        self._remove_file()
-
-    def _remove_file(self):
```

Comment:
Why are you removing this?

Reply:
The actual file deletion is handled by the signal handlers
The FirmwareImage model's delete method is calling _remove_file() after super().delete(), which means it's trying to delete files that the signal handlers are also trying to delete

Reply:
The `delete()` method gets called when an instance of FirmwareImage gets deleted.
However, the delete method is not called when the deletion is cascaded from parent models.

I think this is not right. Are you sure this is right?